### PR TITLE
Answer:49

### PR DESCRIPTION
**Pull request by dungcodejob**

### CODE REVIEW

**Comment by tomalaforge** [comment]
@alcaidio you have your first answer

File: apps/rxjs/hold-to-save-btn/src/app/app.component.ts [hunk 1]
```diff
@@ -1,19 +1,26 @@
+import { AsyncPipe } from '@angular/common';
 import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { HoldableDirective } from './holdable.directive';
 
 @Component({
   standalone: true,
-  imports: [],
+  imports: [HoldableDirective, AsyncPipe],
```
Comment (alcaidio) [inline apps/rxjs/hold-to-save-btn/src/app/app.component.ts:7]: async pipe unused here

File: apps/rxjs/hold-to-save-btn/src/app/app.component.ts [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+          appHoldable
+          #holdable="holdable"
+          (active)="onSend()"
           class="rounded bg-indigo-600 px-4 py-2 font-bold text-white transition-colors ease-in-out hover:bg-indigo-700">
           Hold me
         </button>
 
-        <progress [value]="20" [max]="100"></progress>
+        <progress
+          [value]="holdable.time() / (2000 / 100)"
```
Comment (alcaidio) [inline apps/rxjs/hold-to-save-btn/src/app/app.component.ts:22]: Maybe it's better to manage the conversion in the directive?

File: apps/rxjs/hold-to-save-btn/src/app/holdable.directive.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  map,
+  merge,
+  switchMap,
+  takeUntil,
+  takeWhile,
+} from 'rxjs';
+
+type Ms = number;
+
+const DEFAULT_HOLD_TIME: Ms = 2000;
```
Comment (alcaidio) [inline apps/rxjs/hold-to-save-btn/src/app/holdable.directive.ts:15]: maybe it's better if we can configurate this from the directive?

File: apps/rxjs/hold-to-save-btn/src/app/app.component.ts (hunk 2 above)
Comment (dungcodejob) [reply apps/rxjs/hold-to-save-btn/src/app/app.component.ts]: Got it. I moved it to directive

File: apps/rxjs/hold-to-save-btn/src/app/holdable.directive.ts (hunk 3 above)
Comment (dungcodejob) [reply apps/rxjs/hold-to-save-btn/src/app/holdable.directive.ts]: I allow user to configurate this by provide a value to DEFAULT_HOLD_TIME_TOKEN

**Review by alcaidio — APPROVED** [review]
Maybe a bit overkill to manage the hold time configuration via a provider rather than just a directive input, but it works well. Good job!

**Pull request closed without merging**